File: backend/src/main/java/com/youtube/clone/youtube_clone/dto/VideoDto.java
package com.youtube.clone.youtube_clone.dto;

import java.util.Set;

import com.youtube.clone.youtube_clone.model.VideoStatus;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VideoDto {
    private String id;
    private String title;
    private String description;
    private Set<String> tags;
    private VideoStatus videoStatus;
    private String thumbnailUrl;
    private String videoUrl;
    private Integer likeCount;
    private Integer dislikeCount;
    private Integer viewCount;

}

File: frontend/youtube-clone-ui/src/app/video-dto.ts
export interface VideoDto{
    id: string;
    title: string;
    description: string;
    tags: Array<string>;
    videoStatus: string;
    thumbnailUrl: string;
    videoUrl: string;
    likeCount: number;
    dislikeCount: number;
    viewCount: number;
}